lib/queue.js

```diff
@@ -430,27 +430,26 @@ Queue.prototype.run = function(concurrency){
   var promises = [];
   var _this = this;
 
+  while(concurrency--){
+    promises.push(new Promise(_this.processJobs));
+  }
+
   // In case of connection loss, running `processStalledJobs` will repick jobs here.
   var start = this.opts.processStalledJobs ? this.processStalledJobs() : Promise.resolve();
```

Comment:
This can be simplified. Something like:

```
  if (this.opts.processStalledJobs) {
    if (this.stalledJobsInterval) clearInterval(this.stalledJobsInterval);
    // Run once and then schedule to run on LOCK_RENEW_TIME interval.
    promises.push(this.processStalledJobs().then(function() {
      _this.stalledJobsInterval = setInterval(this.processStalledJobs, _this.LOCK_RENEW_TIME);
    }));
  }

```